impeller/renderer/backend/vulkan/render_pass_vk.cc

```diff
@@ -40,13 +47,256 @@ bool RenderPassVK::OnEncodeCommands(const Context& context) const {
+  res = command_buffer_.end();
+  if (res != vk::Result::eSuccess) {
+    VALIDATION_LOG << "Failed to end command buffer: " << vk::to_string(res);
+    return false;
+  }
+
+  return true;
+}
+
+bool RenderPassVK::EncodeCommand(const Context& context,
+                                 const Command& command) const {
+  SetViewportAndScissor(command);
+
+  auto& pipeline_vk = PipelineVK::Cast(*command.pipeline);
+  PipelineCreateInfoVK* pipeline_create_info = pipeline_vk.GetCreateInfo();
+
+  if (!AllocateAndBindDescriptorSets(context, command, pipeline_create_info)) {
+    return false;
+  }
+
+  command_buffer_.bindPipeline(vk::PipelineBindPoint::eGraphics,
+                               pipeline_create_info->GetVKPipeline());
+
+  auto vertex_buffer_view = command.GetVertexBuffer();
+  auto index_buffer_view = command.index_buffer;
+
+  if (!vertex_buffer_view || !index_buffer_view) {
+    return false;
+  }
+
+  auto& allocator = *context.GetResourceAllocator();
+  const auto& pipeline_desc = command.pipeline->GetDescriptor();
+
+  auto vertex_buffer = vertex_buffer_view.buffer->GetDeviceBuffer(allocator);
+  auto index_buffer = index_buffer_view.buffer->GetDeviceBuffer(allocator);
+
+  if (!vertex_buffer || !index_buffer) {
+    VALIDATION_LOG << "Failed to acquire device buffers"
+                   << " for vertex and index buffer views";
+    return false;
+  }
+
+  // bind vertex buffer
+  auto vertex_buffer_handle =
+      DeviceBufferVK::Cast(*vertex_buffer).GetVKBufferHandle();
+  vk::Buffer vertex_buffers[] = {vertex_buffer_handle};
+  vk::DeviceSize vertex_buffer_offsets[] = {vertex_buffer_view.range.offset};
+  command_buffer_.bindVertexBuffers(0, 1, vertex_buffers,
+                                    vertex_buffer_offsets);
+
+  // index buffer
+  auto index_buffer_handle =
+      DeviceBufferVK::Cast(*index_buffer).GetVKBufferHandle();
+  command_buffer_.bindIndexBuffer(index_buffer_handle,
+                                  index_buffer_view.range.offset,
+                                  ToVKIndexType(command.index_type));
+
+  // execute draw
+  command_buffer_.drawIndexed(command.index_count, command.instance_count, 0, 0,
+                              0);
+  return true;
+}
+
+bool RenderPassVK::AllocateAndBindDescriptorSets(
+    const Context& context,
+    const Command& command,
+    PipelineCreateInfoVK* pipeline_create_info) const {
+  auto& allocator = *context.GetResourceAllocator();
+  vk::PipelineLayout pipeline_layout =
+      pipeline_create_info->GetPipelineLayout();
+
+  const auto& context_vk = ContextVK::Cast(context);
+  const auto& pool = context_vk.GetDescriptorPool();
+
+  vk::DescriptorSetAllocateInfo alloc_info;
+  std::array<vk::DescriptorSetLayout, 1> dsls = {
+      pipeline_create_info->GetDescriptorSetLayout(),
+  };
+
+  alloc_info.setDescriptorPool(pool->GetPool());
+  alloc_info.setSetLayouts(dsls);
+
+  auto desc_sets_res = device_.allocateDescriptorSets(alloc_info);
```

Comment:
These sets are leaked right?

Reply:
Yup, leaked for now. Will make a descriptor pool for render-pass or add these to the list of things that need to be freed post submit.